src/Adapter/AmqpLib/Connection/AmqpConnection.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FiveLab\Component\Amqp\Adapter\AmqpLib\Connection;
+
+use FiveLab\Component\Amqp\Connection\ConnectionInterface;
+use PhpAmqpLib\Connection\AbstractConnection;
+use SplObserver;
+
+/**
+ * The connection provided via php-amqplib library.
+ */
+class AmqpConnection implements ConnectionInterface, \SplSubject
+{
+    /**
+     * @var \SplObserver[]
+     */
+    private $observers = [];
+
+    /**
+     * @var AbstractConnection
+     */
+    private $connection;
+
+    /**
+     * @var float
+     */
+    private $readTimeout;
+
+    /**
+     * @param AbstractConnection $connection
+     * @param float              $readTimeout
+     */
+    public function __construct(AbstractConnection $connection, float $readTimeout)
+    {
+        $this->connection = $connection;
+        $this->readTimeout = $readTimeout;
+    }
+
+    /**
+     * Closes connection
+     * @throws \Exception
+     */
+    public function __destruct()
+    {
+        $this->connection->close();
+    }
+
+    /**
+     * Returns original connection
+     * This method establishes the connection
+     *
+     * @return AbstractConnection
+     */
+    public function getConnection(): AbstractConnection
+    {
+        return $this->connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function connect(): void
+    {
+        if (!$this->connection->isConnected()) {
+            $this->connection->reconnect();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isConnected(): bool
+    {
+        return $this->connection && $this->connection->isConnected();
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Exception
```

Comment:
`@throws \Exception` in this comment is excess

Reply:
interface should know about possible exceptions